pkg/controller/mpi_job_controller.go

```diff
@@ -441,9 +441,22 @@ func (c *MPIJobController) Run(threadiness int, stopCh <-chan struct{}) error {
 	if c.PodGroupCtrl != nil {
 		synced = append(synced, c.podGroupSynced, c.priorityClassSynced)
 	}
-	if ok := cache.WaitForCacheSync(stopCh, synced...); !ok {
+
+	timeout, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
What if you have so many jobs that 30s are not enough for the cache to sync?

Reply:
Changed the timeout duration to an option / flag. 

Not sure if that truly addresses the underlying question - but at least it's now configurable ...